client/components/form-sections/application-table.tsx

```diff
@@ -1,141 +1,171 @@
-  const indexOfLastApplication = currentPage * applicationsPerPage;
-  const indexOfFirstApplication = indexOfLastApplication - applicationsPerPage;
-  const currentApplications = filteredApplications.slice(
-    indexOfFirstApplication,
-    indexOfLastApplication,
-  );
+  useEffect(() => {
+    (async () => {
+      if (Object.keys(query).length === 0) return;
+      setApplicationData(query);
+    })();
+  }, [query]);
 
   // Change pages
   const nextPage = () => {
-    if (currentPage == Math.floor(filteredApplications.length / applicationsPerPage) + 1) return;
-    setCurrentPage(currentPage + 1);
+    if (
+      !totalApplications ||
+      Number(query.page) == Math.ceil(totalApplications / Number(query.limit))
+    )
+      return;
+    push(
+      { query: { ...query, page: Number(query.page) + 1, limit: Number(query.limit) } },
+      undefined,
+      { shallow: true },
+    );
   };
   const previousPage = () => {
-    if (currentPage == 1) return;
-    setCurrentPage(currentPage - 1);
+    if (Number(query.page) == 1) return;
+    push(
+      { query: { ...query, page: Number(query.page) - 1, limit: Number(query.limit) } },
+      undefined,
+      { shallow: true },
+    );
   };
   const firstPage = () => {
-    if (currentPage == 1) return;
-    setCurrentPage(1);
+    if (Number(query.page) == 1) return;
+    push({ query: { ...query, page: 1, limit: Number(query.limit) } }, undefined, {
+      shallow: true,
+    });
   };
   const lastPage = () => {
-    if (currentPage == filteredApplications.length / applicationsPerPage) return;
-    setCurrentPage(Math.floor(filteredApplications.length / applicationsPerPage) + 1);
+    if (
+      !totalApplications ||
+      Number(query.page) == Math.ceil(totalApplications / Number(query.limit))
+    )
+      return;
+    push(
+      {
+        query: {
+          ...query,
+          page: Math.ceil(totalApplications / Number(query.page)),
+          limit: Number(query.limit),
+        },
+      },
+      undefined,
+      { shallow: true },
+    );
+  };
+
+  const checkInputs = searchFacilityName.length == 0 && searchConfirmatioID.length == 0;
+  const handleFilter = () => {
+    if (checkInputs) return;
+    push(
+      {
+        query: { ...query, facilityName: searchFacilityName, confirmationId: searchConfirmatioID },
+      },
+      undefined,
+      { shallow: true },
+    );
+  };
+
+  const handleClear = () => {
+    if (checkInputs) return;
+    // Clear Inputs
+    setState(state => ({ ...state, searchFacilityName: '', searchConfirmatioID: '' }));
+
+    push(
+      { query: { ...query, facilityName: '', confirmationId: '', limit: Number(query.limit) } },
+      undefined,
+      { shallow: true },
+    );
   };
 
   return (
     <div>
       <div className='w-full bg-white flex my-2 justify-between'>
         <h1 className='text-2xl font-bold h-6 text-bcBluePrimary text-left flex-col items-start'>
-          Applications
+          Applications {query.limit}
```

Comment:
Shouldn't directly use the query.limit. Should have a common parser, that'll parse the query and will return limit, page and so on!